lib/engine/route.rb

```diff
@@ -123,16 +123,22 @@ def touch_node(node)
             connection_data << segment(connection, left: tail[:right])
           end
         end
+        connection_data.pop if @train.local? && connection_data.size == 2
```

Comment:
why do you pop off a connection if the data is 2? 

how can you ever run a city to a town?

Reply:
Yes you can the rules states as following:

L (local) trains run in a city which has a station token of the owning (minor or major) company. They can additionally run to a single small station, but are not required to do so. They can thus be considered 1 (+1) trains. And if you run the lokal to a town you get a correct connection object. Remove the one created when you just pressed the city.